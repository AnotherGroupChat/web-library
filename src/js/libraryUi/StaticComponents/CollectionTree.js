'use strict';

var log = require('libzotero/lib/Log').Logger('zotero-web-library:CollectionTree');

var React = require('react');

var CollectionTree = React.createClass({
	getDefaultProps: function() {
		return {};
	},
	getInitialState: function() {
		return {};
	},
	render: function() {
		return (
			<nav id="collection-tree">
				<h3 className="offscreen">Library navigation</h3>
				<section>
					<h4>My library</h4>
					<ul>
						<li className="active"><a href="#">All documents</a></li>
						<li><a href="#">Collection 1</a></li>
						<li><a href="#">Collection 2</a></li>
						<li><a href="#">Collection 3</a></li>
					</ul>
				</section>
				<section>
					<h4>Group libraries</h4>
					<ul>
						<li><a href="#">Group library 1</a></li>
						<li><a href="#">Group library 2</a></li>
						<li><a href="#">Group library 3</a></li>
					</ul>
				</section>
			</nav>
		);
	}
});

module.exports = CollectionTree;
