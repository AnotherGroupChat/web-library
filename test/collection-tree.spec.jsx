import React from 'react';
import renderer from 'react-test-renderer';

import CollectionTree from '../src/js/component/collection-tree';
import collectionsFix from './fixtures/collections';

const mapCollectionsRecursive = (col, f) => {
	return col.map(c => {
		c = f(c);
		if(c.children.length) {
			c.children = mapCollectionsRecursive(c.children, f);
		}
		return c;
	});
};

it('Renders collapsed', () => {
	let collections = mapCollectionsRecursive(collectionsFix, c => ({
		isOpen: false,
		isSelected: false,
		...c
	}));
	let tree = renderer.create(
		<CollectionTree collections = { collections } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders expanded', () => {
	let collections = mapCollectionsRecursive(collectionsFix, c => ({
		isOpen: true,
		isSelected: false,
		...c
	}));
	let tree = renderer.create(
		<CollectionTree collections = { collections } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});