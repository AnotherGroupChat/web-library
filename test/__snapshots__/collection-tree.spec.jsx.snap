// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders collapsed 1`] = `
<nav
  className="collection-tree"
>
  <header
    className="touch-header hidden-mouse-md-up hidden-xs-down"
  >
    <h3>
      Library
    </h3>
  </header>
  <div
    className="level-root active"
  >
    <div
      className="scroll-container"
      role="tree"
    >
      <section>
        <h4>
          My Library
        </h4>
        <div
          className="level level-1  "
        >
          <ul
            className="nav"
            role="group"
          >
            <li
              className=" "
            >
              <div
                aria-expanded={false}
                className="item-container"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="treeitem"
                tabIndex="0"
              >
                <div
                  className="twisty-container"
                >
                  <button
                    className="twisty"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    type="button"
                  />
                </div>
                <svg
                  className="svg-icon icon-folders touch"
                  role="img"
                  style={
                    Object {
                      "height": "28px",
                      "width": "28px",
                    }
                  }
                >
                  <use
                    viewBox="0 0 512 512"
                    xlinkHref="/icons/28/folders.svg#folders"
                  />
                </svg>
                <svg
                  className="svg-icon icon-folder mouse"
                  role="img"
                  style={
                    Object {
                      "height": "16px",
                      "width": "16px",
                    }
                  }
                >
                  <use
                    viewBox="0 0 512 512"
                    xlinkHref="/icons/16/folder.svg#folder"
                  />
                </svg>
                <a>
                  depth-1
                </a>
              </div>
              <div
                className="level level-2  "
              >
                <ul
                  className="nav"
                  role="group"
                >
                  <li
                    className=" "
                  >
                    <div
                      aria-expanded={false}
                      className="item-container"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="treeitem"
                      tabIndex="0"
                    >
                      <div
                        className="twisty-container"
                      >
                        <button
                          className="twisty"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          type="button"
                        />
                      </div>
                      <svg
                        className="svg-icon icon-folders touch"
                        role="img"
                        style={
                          Object {
                            "height": "28px",
                            "width": "28px",
                          }
                        }
                      >
                        <use
                          viewBox="0 0 512 512"
                          xlinkHref="/icons/28/folders.svg#folders"
                        />
                      </svg>
                      <svg
                        className="svg-icon icon-folder mouse"
                        role="img"
                        style={
                          Object {
                            "height": "16px",
                            "width": "16px",
                          }
                        }
                      >
                        <use
                          viewBox="0 0 512 512"
                          xlinkHref="/icons/16/folder.svg#folder"
                        />
                      </svg>
                      <a>
                        depth-2
                      </a>
                    </div>
                    <div
                      className="level level-3  "
                    >
                      <ul
                        className="nav"
                        role="group"
                      >
                        <li
                          className=" "
                        >
                          <div
                            aria-expanded={false}
                            className="item-container"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            role="treeitem"
                            tabIndex="0"
                          >
                            <div
                              className="twisty-container"
                            >
                              <button
                                className="twisty"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                type="button"
                              />
                            </div>
                            <svg
                              className="svg-icon icon-folders touch"
                              role="img"
                              style={
                                Object {
                                  "height": "28px",
                                  "width": "28px",
                                }
                              }
                            >
                              <use
                                viewBox="0 0 512 512"
                                xlinkHref="/icons/28/folders.svg#folders"
                              />
                            </svg>
                            <svg
                              className="svg-icon icon-folder mouse"
                              role="img"
                              style={
                                Object {
                                  "height": "16px",
                                  "width": "16px",
                                }
                              }
                            >
                              <use
                                viewBox="0 0 512 512"
                                xlinkHref="/icons/16/folder.svg#folder"
                              />
                            </svg>
                            <a>
                              depth-3
                            </a>
                          </div>
                          <div
                            className="level level-4  "
                          >
                            <ul
                              className="nav"
                              role="group"
                            >
                              <li
                                className=" "
                              >
                                <div
                                  aria-expanded={false}
                                  className="item-container"
                                  onClick={[Function]}
                                  onKeyPress={[Function]}
                                  role="treeitem"
                                  tabIndex="0"
                                >
                                  <div
                                    className="twisty-container"
                                  >
                                    <button
                                      className="twisty"
                                      onClick={[Function]}
                                      onKeyPress={[Function]}
                                      type="button"
                                    />
                                  </div>
                                  <svg
                                    className="svg-icon icon-folder touch"
                                    role="img"
                                    style={
                                      Object {
                                        "height": "28px",
                                        "width": "28px",
                                      }
                                    }
                                  >
                                    <use
                                      viewBox="0 0 512 512"
                                      xlinkHref="/icons/28/folder.svg#folder"
                                    />
                                  </svg>
                                  <svg
                                    className="svg-icon icon-folder mouse"
                                    role="img"
                                    style={
                                      Object {
                                        "height": "16px",
                                        "width": "16px",
                                      }
                                    }
                                  >
                                    <use
                                      viewBox="0 0 512 512"
                                      xlinkHref="/icons/16/folder.svg#folder"
                                    />
                                  </svg>
                                  <a>
                                    depth-4
                                  </a>
                                </div>
                              </li>
                            </ul>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </section>
      <section>
        <h4>
          Group Libraries
        </h4>
      </section>
    </div>
  </div>
</nav>
`;

exports[`Renders expanded 1`] = `
<nav
  className="collection-tree"
>
  <header
    className="touch-header hidden-mouse-md-up hidden-xs-down"
  >
    <h3>
      Library
    </h3>
  </header>
  <div
    className="level-root active"
  >
    <div
      className="scroll-container"
      role="tree"
    >
      <section>
        <h4>
          My Library
        </h4>
        <div
          className="level level-1  "
        >
          <ul
            className="nav"
            role="group"
          >
            <li
              className="open "
            >
              <div
                aria-expanded={true}
                className="item-container"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="treeitem"
                tabIndex="0"
              >
                <div
                  className="twisty-container"
                >
                  <button
                    className="twisty"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    type="button"
                  />
                </div>
                <svg
                  className="svg-icon icon-folders touch"
                  role="img"
                  style={
                    Object {
                      "height": "28px",
                      "width": "28px",
                    }
                  }
                >
                  <use
                    viewBox="0 0 512 512"
                    xlinkHref="/icons/28/folders.svg#folders"
                  />
                </svg>
                <svg
                  className="svg-icon icon-folder mouse"
                  role="img"
                  style={
                    Object {
                      "height": "16px",
                      "width": "16px",
                    }
                  }
                >
                  <use
                    viewBox="0 0 512 512"
                    xlinkHref="/icons/16/folder.svg#folder"
                  />
                </svg>
                <a>
                  depth-1
                </a>
              </div>
              <div
                className="level level-2  "
              >
                <ul
                  className="nav"
                  role="group"
                >
                  <li
                    className="open "
                  >
                    <div
                      aria-expanded={true}
                      className="item-container"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="treeitem"
                      tabIndex="0"
                    >
                      <div
                        className="twisty-container"
                      >
                        <button
                          className="twisty"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          type="button"
                        />
                      </div>
                      <svg
                        className="svg-icon icon-folders touch"
                        role="img"
                        style={
                          Object {
                            "height": "28px",
                            "width": "28px",
                          }
                        }
                      >
                        <use
                          viewBox="0 0 512 512"
                          xlinkHref="/icons/28/folders.svg#folders"
                        />
                      </svg>
                      <svg
                        className="svg-icon icon-folder mouse"
                        role="img"
                        style={
                          Object {
                            "height": "16px",
                            "width": "16px",
                          }
                        }
                      >
                        <use
                          viewBox="0 0 512 512"
                          xlinkHref="/icons/16/folder.svg#folder"
                        />
                      </svg>
                      <a>
                        depth-2
                      </a>
                    </div>
                    <div
                      className="level level-3  "
                    >
                      <ul
                        className="nav"
                        role="group"
                      >
                        <li
                          className="open "
                        >
                          <div
                            aria-expanded={true}
                            className="item-container"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            role="treeitem"
                            tabIndex="0"
                          >
                            <div
                              className="twisty-container"
                            >
                              <button
                                className="twisty"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                type="button"
                              />
                            </div>
                            <svg
                              className="svg-icon icon-folders touch"
                              role="img"
                              style={
                                Object {
                                  "height": "28px",
                                  "width": "28px",
                                }
                              }
                            >
                              <use
                                viewBox="0 0 512 512"
                                xlinkHref="/icons/28/folders.svg#folders"
                              />
                            </svg>
                            <svg
                              className="svg-icon icon-folder mouse"
                              role="img"
                              style={
                                Object {
                                  "height": "16px",
                                  "width": "16px",
                                }
                              }
                            >
                              <use
                                viewBox="0 0 512 512"
                                xlinkHref="/icons/16/folder.svg#folder"
                              />
                            </svg>
                            <a>
                              depth-3
                            </a>
                          </div>
                          <div
                            className="level level-4  "
                          >
                            <ul
                              className="nav"
                              role="group"
                            >
                              <li
                                className="open "
                              >
                                <div
                                  aria-expanded={true}
                                  className="item-container"
                                  onClick={[Function]}
                                  onKeyPress={[Function]}
                                  role="treeitem"
                                  tabIndex="0"
                                >
                                  <div
                                    className="twisty-container"
                                  >
                                    <button
                                      className="twisty"
                                      onClick={[Function]}
                                      onKeyPress={[Function]}
                                      type="button"
                                    />
                                  </div>
                                  <svg
                                    className="svg-icon icon-folder touch"
                                    role="img"
                                    style={
                                      Object {
                                        "height": "28px",
                                        "width": "28px",
                                      }
                                    }
                                  >
                                    <use
                                      viewBox="0 0 512 512"
                                      xlinkHref="/icons/28/folder.svg#folder"
                                    />
                                  </svg>
                                  <svg
                                    className="svg-icon icon-folder mouse"
                                    role="img"
                                    style={
                                      Object {
                                        "height": "16px",
                                        "width": "16px",
                                      }
                                    }
                                  >
                                    <use
                                      viewBox="0 0 512 512"
                                      xlinkHref="/icons/16/folder.svg#folder"
                                    />
                                  </svg>
                                  <a>
                                    depth-4
                                  </a>
                                </div>
                              </li>
                            </ul>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </section>
      <section>
        <h4>
          Group Libraries
        </h4>
      </section>
    </div>
  </div>
</nav>
`;
